Tests/test_GenBank_unittest.py

```diff
@@ -432,6 +434,102 @@ def test_genbank_date_invalid(self):
             gb = SeqIO.read(handle, "gb")
             self.assertEqual(gb.annotations["date"], "01-JAN-1980")
 
+    def test_longer_locus_line(self):
+        """Check that we can read and write files with longer locus lines"""
+        # Create example file from existing file
+        with open(path.join("GenBank", "DS830848.gb"), 'r') as inhandle:
+            data = inhandle.readlines()
+            data[0] = "LOCUS       AZZZAA021234567891234 2147483647 bp    DNA     linear   PRI 15-OCT-2018\n"
+
+        # Create memory file from modified genbank file
+        in_tmp = StringIO()
+        in_tmp.writelines(data)
+        in_tmp.seek(0)
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("error", BiopythonParserWarning)
+            try:
+                record = SeqIO.read(in_tmp, 'genbank')
+            except BiopythonParserWarning as e:
+                self.assertEqual(str(e), "Attempting to parse locus line that is extra long or malformed  :\n"
+                                 "'LOCUS       AZZZAA021234567891234 2147483647 bp    DNA     linear   PRI 15-OCT-2018\\n'\n"
+                                 "Found locus 'AZZZAA021234567891234' size '2147483647' residue_type 'DNA'\n"
+                                 "Some fields may be wrong.")
+            else:
+                self.assertTrue(False, "Expected specified BiopythonParserWarning here.")
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            in_tmp.seek(0)
+            record = SeqIO.read(in_tmp, 'genbank')
+
+            # Create temporary output memory file
+            out_tmp = StringIO()
+            SeqIO.write(record, out_tmp, 'genbank')
+
+            # Check that the written file can be read back in
+            out_tmp.seek(0)
+            record_in = SeqIO.read(out_tmp, 'genbank')
+            self.assertEqual(record_in.id, "DS830848.1")
+            self.assertEqual(record_in.name, "AZZZAA021234567891234")
+            self.assertEqual(len(record_in.seq), 2147483647)
+
+    if sys.maxsize > 2147483647:
+        def test_extremely_long_sequence(self):
+            """Tests if extremely long sequences can be read.
+
+            This is only run if sys.maxsize > 2147483647.
```

Comment:
Can you add a note here about Python 2, it would help for removing all the Python 2 specific hacks in 2020 if we can find it with ``grep -i "Python 2"`` or ``grep -i "Python 3"``

Reply:
As far as I can tell, this will still be a limitation in 32-bit Python3. See the documentation [here ](https://docs.python.org/3/reference/datamodel.html#object.__len__) and [here](https://docs.python.org/3/library/sys.html#sys.maxsize). This is why I decided to have the conditional statement based on the `sys.maxsize` rather than the python version.

Reply:
OK, good point. I've not knowingly used a 32bit Python for some time, so that didn't cross my mind.